src/Serilog/Formatting/Json/JsonValueFormatter.cs

```diff
@@ -221,10 +227,27 @@ static void FormatBooleanValue(bool value, TextWriter output)
             output.Write(value ? "true" : "false");
         }
 
-        static void FormatApproximateNumericValue(IFormattable value, TextWriter output)
-        {
-            output.Write(value.ToString("R", CultureInfo.InvariantCulture));
-        }
+		static void FormatFloatValue(float value, TextWriter output)
+		{
+			if (float.IsNaN(value) || float.IsInfinity(value))
+			{
+				FormatStringValue(value.ToString(), output);
```

Comment:
Just checked out how double behaves in different cultures; I think for this to work conveniently we should use `ToString(CultureInfo.InvariantCulture)` here.